src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1170,6 +1174,26 @@ private function getDynamicParameter(\$name)
 {$getDynamicParameter}
     }
 
+
+EOF;
+
+            $code .= '    private $normalizedParameterNames = '.($normalizedParams ? sprintf("array(\n%s\n    );", implode("\n", $normalizedParams)) : 'array();')."\n";
+            $code .= <<<'EOF'
+
+    private function normalizeParameterName($name)
+    {
+        if (isset($this->normalizedParameterNames[$normalizedName = strtolower($name)]) || isset($this->parameters[$normalizedName]) || array_key_exists($normalizedName, $this->parameters)) {
+            $normalizedName = isset($this->normalizedParameterNames[$normalizedName]) ? $this->normalizedParameterNames[$normalizedName] : $normalizedName;
```

Comment:
that is not the normalized (lowercased) parameter name, but the original name. using the same variable for both things is confusing